### PR TITLE
Fix TS - make GlobalConfig type public

### PR DESCRIPTION
**Pull request by ivanblinov2k17**

### CODE REVIEW

File: js/core/config.d.ts [hunk 1]
```diff
@@ -25,15 +25,16 @@ declare function config(): globalConfig;
...
 
 /**
  * @docid
  * @section commonObjectStructures
  * @namespace DevExpress
+ * @public
  * @type object
  */
-export interface globalConfig {
+export interface GlobalConfig {
```
Comment (IlyaKhD) [inline js/core/config.d.ts]: Add alias for backward compatibility
```ts
/**
 * @deprecated Use GlobalConfig instead
 */
export globalConfig = GlobalConfig;
 ```

**Review by IlyaKhD — CHANGES_REQUESTED** [review]

File: js/core/config.d.ts [hunk 2]
```diff
@@ -146,4 +147,10 @@ export interface globalConfig {
     useLegacyVisibleIndex?: boolean;
 }
 
+/**
+ * @docid
```
Comment (IlyaKhD) [inline js/core/config.d.ts]: This one doesn't need `docid` tag because it doesn't have it's own topic (`GlobalConfig` should have)
```suggestion
```

**Review by IlyaKhD — CHANGES_REQUESTED** [review]

File: js/core/config.d.ts [hunk 3]
```diff
@@ -146,4 +147,10 @@ export interface globalConfig {
...
+ * @deprecated Use GlobalConfig instead
+ */
+export interface globalConfig extends GlobalConfig {}
```
Comment (IlyaKhD) [inline js/core/config.d.ts]:
```suggestion
export type globalConfig = GlobalConfig;
```

File: js/core/config.d.ts (hunk 1 above)
Comment (IlyaKhD) [inline js/core/config.d.ts]:
```suggestion
export type GlobalConfig = {
```

File: ts/aliases.d.ts [hunk 4]
```diff
@@ -425,6 +425,11 @@ declare module DevExpress.common.charts {
 
 // v22.2
 
+declare module DevExpress {
+    /** @deprecated Use GlobalConfig instead */
```
Comment (IlyaKhD) [inline ts/aliases.d.ts]:
```suggestion
    /** @deprecated Use DevExpress.GlobalConfig instead */
```

**Review by IlyaKhD — DISMISSED** [review]

File: js/core/config.d.ts [hunk 5]
```diff
@@ -25,15 +25,16 @@ declare function config(): globalConfig;
...
  */
 // eslint-disable-next-line @typescript-eslint/no-shadow
-declare function config(config: globalConfig): void;
+declare function config(config: GlobalConfig): void;
 
 /**
  * @docid
  * @section commonObjectStructures
  * @namespace DevExpress
+ * @public
```
Comment (IlyaKhD) [inline js/core/config.d.ts:34]: Let's change it's namespace as well
```diff
-* @namespace DevExpress
+* @namespace DevExpress.common
```

**Review by vladaskorohodova — DISMISSED** [review]

File: ts/aliases.d.ts [hunk 6]
```diff
@@ -425,6 +425,11 @@ declare module DevExpress.common.charts {
 
 // v22.2
 
+declare module DevExpress.common {
```
Comment (IlyaKhD) [inline ts/aliases.d.ts]:
```suggestion
declare module DevExpress {
```

**Review by IlyaKhD — CHANGES_REQUESTED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Review by vladaskorohodova — APPROVED** [review]

**Pull request merged**